src/application.c

```diff
@@ -255,6 +255,12 @@ bool app_init(const struct config* cfg, const char* const* sources, size_t num)
                                  ARRAY_SIZE(mode_names));
     ctx.mprev = mode_viewer;
 
+    // set mode to gallery if current mode is not default and
+    // the imglist is multiple images for expected usage
+    if (imglist_size() > 1 && ctx.mcurr != ctx.mprev) {
```

Comment:
This makes the `general.mode` option useless and adds a few questions:
1. How to start the `viewer` mode with multiple files?
2. How to start slide show?

Also, the image list at this moment contains a list of _files_, not _images_.
In fact, the list could contain hundreds of entries, but only one file with image data.

Reply:
Your concerns are correct. There is no way to check if the current mode was explicitly set by the user and not defaults. I have no workaround for this.

> Also, the image list at this moment contains a list of files, not images.
> In fact, the list could contain hundreds of entries, but only one file with image data.

Sorry, but what exactly is the reasoning for this? swayimg is unable to load the files that aren't images, so why are they remembered in memory?